pipeline/tests/pil.rs

```diff
@@ -319,14 +325,11 @@ fn serialize_deserialize_optimized_pil() {
     assert_eq!(input_pil_file, output_pil_file);
 }
 
-// TODO re-enable when we have a working backend.
-// #[test]
-// fn challenges() {
-//     let f = "pil/challenges.pil";
-//     verify_pil(f, Default::default());
-//     test_halo2(f, Default::default());
-//     gen_estark_proof(f, Default::default());
-// }
+#[test]
+fn challenges() {
+    let f = "pil/challenges.pil";
+    test_halo2(f, Default::default());
```

Comment:
I left out the Goldilocks provers! I'm wondering if we still need that test though? It doesn't really test anything that is not tested by `permutation_via_challenges`, right? And `permutation_via_challenges` is a real-world use-case.

Reply:
It's fine, but then please also delete the `pil/challenges.pil` file because it is unused.